### PR TITLE
Cleanup Jobs for additional namespaces

### PR DESCRIPTION
**Pull request by jala-dx**

### CODE REVIEW

File: kotsadm/operator/pkg/client/client.go [hunk 1]
```diff
@@ -268,6 +269,10 @@ func (c *Client) registerHandlers(socketClient *socket.Client) error {
 				// we don't fail here...
 				log.Printf("error creating namespace: %s", deployError.Error())
 			}
+			if _, ok := c.ExistingInformers[additionalNamespace]; !ok {
+				c.runHooksInformer(additionalNamespace)
```
Comment (divolgin) [inline kotsadm/operator/pkg/client/client.go:273]: missing error check

File: kotsadm/operator/pkg/client/hooks.go [hunk 2]
```diff
@@ -32,12 +32,13 @@ func (c *Client) runHooksInformer() error {
```
Comment (divolgin) [inline kotsadm/operator/pkg/client/hooks.go:32]: line 31 wipes out `c.hookStopChans` on the second call

File: kotsadm/pkg/socketservice/socketservice.go [hunk 3]
```diff
@@ -328,6 +328,8 @@ func processDeploySocketForApp(clusterSocket *ClusterSocket, a *apptypes.App) er
 	if err != nil {
 		return errors.Wrap(err, "failed to get socket channel from server")
 	}
+	// Event is sent here
+	fmt.Println("kotsadm: received deploy event", deployArgs.AdditionalNamespaces)
```
Comment (divolgin) [inline kotsadm/pkg/socketservice/socketservice.go:332]: there is no need for this log and comment

File: kotsadm/operator/pkg/client/client.go [hunk 4]
```diff
@@ -85,7 +86,7 @@ func (c *Client) Run() error {
 
 	supportbundle.StartServer()
 
-	c.runHooksInformer()
+	c.runHooksInformer(c.TargetNamespace)
```
Comment (divolgin) [inline kotsadm/operator/pkg/client/client.go:89]: How about adding `TargetNamespace` to `ExistingInformers` as well for consistency.

**Review by divolgin — CHANGES_REQUESTED** [review]

File: kotsadm/operator/pkg/client/client.go (hunk 4 above)
Comment (jala-dx) [reply kotsadm/operator/pkg/client/client.go]: done

File: kotsadm/operator/pkg/client/client.go [hunk 5]
```diff
@@ -85,9 +86,15 @@ func (c *Client) Run() error {
 
 	supportbundle.StartServer()
 
-	c.runHooksInformer()
-	defer c.shutdownHooksInformer()
+	if _, ok := c.ExistingInformers[c.TargetNamespace]; !ok {
+		c.ExistingInformers[c.TargetNamespace] = true
+		if err := c.runHooksInformer(c.TargetNamespace); err != nil {
+			// we don't fail here...
+			log.Printf("error registering cleanup hooks for TargetNamespace: %s", err.Error())
```
Comment (divolgin) [inline kotsadm/operator/pkg/client/client.go:93]: lets %s that TargetNamespace to get the actual value logged.

File: kotsadm/operator/pkg/client/client.go [hunk 6]
```diff
@@ -268,6 +275,13 @@ func (c *Client) registerHandlers(socketClient *socket.Client) error {
 				// we don't fail here...
 				log.Printf("error creating namespace: %s", deployError.Error())
 			}
+			if _, ok := c.ExistingInformers[additionalNamespace]; !ok {
+				c.ExistingInformers[additionalNamespace] = true
+				if deployError = c.runHooksInformer(additionalNamespace); deployError != nil {
+					// we don't fail here...
+					log.Printf("error registering cleanup hooks for additionalNamespace: %s", deployError.Error())
```
Comment (divolgin) [inline kotsadm/operator/pkg/client/client.go:282]: same, use %s for `additionalNamespace` to log the actual name

File: kotsadm/operator/pkg/client/client.go (hunk 1 above)
Comment (jala-dx) [reply kotsadm/operator/pkg/client/client.go]: fixed

**Review by divolgin — APPROVED** [review]

**Pull request merged**